lib/src/js/polymer_base_dart.html

```diff
@@ -91,10 +105,32 @@
         // If you call notifyPath from dart we need to do manual modifications
         // on any JsArray or JsObject instances.
         var parts = this._getPathParts(path);
-        if (parts.length > 1) {
-          var model = this.get(parts.splice(0, parts.length - 1));
-          if (model[parts[0]] != value) model[parts[0]] = value;
+        // if (parts.length > 1) {
+
+        var thisArg =  this;
+        for (var i=0;i<parts.length-1;i++) {
```

Comment:
This is definitely going to be a bit weird because when calling `notifyPath` directly from dart it will be the original, unmodified path. However if this is called as a result of calling `set` from dart then you get the modified path.